src/pools/selection-strategies.ts

```diff
@@ -94,7 +98,7 @@ class LessRecentlyUsedWorkerChoiceStrategy<
     // A worker is always found because it picks the one with fewer tasks
     let lessRecentlyUsedWorker!: Worker
     for (const [worker, numberOfTasks] of this.pool.tasks) {
-      if (numberOfTasks === 0) {
+      if (!this.pool.isDynamic() && numberOfTasks === 0) {
```

Comment:
We may extract this `!this.pool.isDynamic()` out of the loop to a `const` and safe performance this way, cause the function is not called for each task